File: JS/circularArrayRotation.js
function circularArrayRotation(a, k, queries) {
    const rotations = k%(a.length) //get only incomplete rotation

    if(rotations!==0) {
        for (let i=0; i<rotations; i++){
            a.unshift(a.pop())
        }
    }

    return queries.map((id) => a[id])
}

console.log(circularArrayRotation([1,2,3, 4], 9, [0,1,2]))

File: Python/circular_array_rotation.py
'''
circularArrayRotation has the following parameter(s):
    int a[n]: the array to rotate
    int k: the rotation count
    int queries[1]: the indices to report
 Returns: int[q]: the values in the rotated a as requested in m
'''
def circular_array_rotation(a, k, queries):
    #get only incomplete rotation
    rotations = k%len(a) 

    #put last item at the beginning ("rotations" times)
    [a.insert(0, a.pop()) for _ in range (rotations)]
        
    return [a[id] for id in queries ]

#example
print(circular_array_rotation([1,2,3], 9, [0,1,2]))
